resources/api/src/main/java/org/jboss/forge/addon/resource/AbstractFileResource.java

```diff
@@ -388,4 +388,18 @@ public OutputStream getResourceOutputStream()
          throw new ResourceException("Error while creating OutputStream for Resource " + this, ioe);
       }
    }
+
+   @Override
+   public void move(FileResource<?> target)
+   {
+      try
+      {
+         getFileOperations().move(file, target.getUnderlyingResourceObject());
+         this.file = target.getUnderlyingResourceObject();
```

Comment:
This won't work if the target is a DirectoryResource. Return the File object from getFileOperation().move() (Change the signature)